src/client/Main.ts

```diff
@@ -39,26 +40,90 @@ class Client {
         document.addEventListener('leave-lobby', this.handleLeaveLobby.bind(this));
         document.addEventListener('single-player', this.handleSinglePlayer.bind(this));
 
-
         const spModal = document.querySelector('single-player-modal') as SinglePlayerModal;
         spModal instanceof SinglePlayerModal
-        document.getElementById('single-player').addEventListener('click', () => {
-            spModal.open();
-        })
+        document.getElementById('single-player').addEventListener('click', async () => {
+            const username = this.usernameInput?.getCurrentUsername();
+
+            if (!username) {
+                usernameValidation.textContent = 'Username is required';
+                return;
+            }
+
+            const isValid = await this.validateUsername(username);
+            if (isValid) {
+                spModal.open();
+            } else {
+                return;
+            }
+        });
 
         const hostModal = document.querySelector('host-lobby-modal') as HostPrivateLobbyModal;
         hostModal instanceof HostPrivateLobbyModal
-        document.getElementById('host-lobby-button').addEventListener('click', () => {
-            hostModal.open();
-        })
+        document.getElementById('host-lobby-button').addEventListener('click', async () => {
+            const username = this.usernameInput?.getCurrentUsername();
+
+            if (!username) {
+                usernameValidation.textContent = 'Username is required';
+                return;
+            }
+
+            const isValid = await this.validateUsername(username);
+
+            if (isValid) {
+                hostModal.open();
+            } else {
+                return;
+            }
+        });
 
         this.joinModal = document.querySelector('join-private-lobby-modal') as JoinPrivateLobbyModal;
         this.joinModal instanceof JoinPrivateLobbyModal
-        document.getElementById('join-private-lobby-button').addEventListener('click', () => {
-            this.joinModal.open();
-        })
+
+        document.getElementById('join-private-lobby-button').addEventListener('click', async () => {
+            const username = this.usernameInput?.getCurrentUsername();
+
+            if (!username) {
+                usernameValidation.textContent = 'Username is required';
+                return;
+            }
+
+            const isValid = await this.validateUsername(username);
+
+            if (isValid) {
+                this.joinModal.open();
+            }else {
+                return;
+            }
+        });
+    }
+
+    private async validateUsername(username: string): Promise<boolean> {
```

Comment:
instead of posting to the server for validation just call the username validator directly, that way we can save a rest call.

Reply:
sure its done